adapters/http-vertx-base/src/test/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapterTest.java

```diff
@@ -410,6 +411,42 @@ public void testUploadCommandResponseFailsForDisabledTenant() {
         verify(metrics, never()).incrementCommandResponseDeliveredToApplication(eq("tenant"));
     }
 
+    /**
+     * Verifies that the adapter fails the upload of a command response with a 403 if the device is not registered and
+     * the valid credentials for the device is available.
+     */
+    @Test
+    public void testUploadCommandResponseFailsForDeletedDevices() {
```

Comment:
the generic case that the device does not exist, isn't it?
So, I would propose to name this method *testUploadCommandResponseFailsForNonExistingDevice*

Reply:
Done